include/envoy/stats/symbol_table.h

```diff
@@ -183,6 +221,13 @@ class SymbolTable {
    *
    */
   virtual StoragePtr encode(absl::string_view name) PURE;
+
+  /**
+   * Called by StatNameSet's destructor.
+   *
+   * @param stat_name_set the set.
+   */
+  virtual void forgetSet(StatNameSet& stat_name_set) PURE;
```

Comment:
Can this be just implicit inside the created stat set so we don't need this at the interface level? It seems like the created implementation of the set can have an explicit handle to the parent and just remove itself?